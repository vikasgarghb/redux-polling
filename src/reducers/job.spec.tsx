import reducer, { defaultJob } from './job';
import {
  ON_ERROR,
  ON_SUCCESS,
  JobStatus,
  ON_CREATE_JOB,
  ON_GET_STATUS,
  ON_COMPLETE_PROCESSING,
  ON_SUCCESS_GET_STATUS,
} from '../actions/job.types';

describe('Job Reducer', () => {
  it('should return default state', () => {
    expect(reducer(undefined, {})).toEqual(defaultJob);
  });

  it('should handle ON_ERROR action', () => {
    expect(reducer(defaultJob, { type: ON_ERROR })).toMatchSnapshot();
  });

  it('should handle ON_GET_STATUS action', () => {
    expect(reducer(defaultJob, { type: ON_GET_STATUS })).toMatchSnapshot();
  });

  it('should handle ON_COMPLETE_PROCESSING action', () => {
    expect(reducer(defaultJob, { type: ON_COMPLETE_PROCESSING })).toMatchSnapshot();
  });

  it('should handle ON_SUCCESS action', () => {
    expect(
      reducer(defaultJob, {
        type: ON_SUCCESS,
        payload: { id: '123', status: JobStatus.Created },
      }),
    ).toMatchSnapshot();
  });

  it('should handle ON_CREATE_JOB action', () => {
    expect(reducer(defaultJob, { type: ON_CREATE_JOB })).toMatchSnapshot();
  });

  it('should handle ON_SUCCESS_GET_STATUS action', () => {
    expect(
      reducer(defaultJob, {
        type: ON_SUCCESS_GET_STATUS,
        payload: { status: JobStatus.Processing },
      }),
    ).toMatchSnapshot();
  });
});
