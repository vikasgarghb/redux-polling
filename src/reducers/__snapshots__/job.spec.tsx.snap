// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job Reducer should handle ON_COMPLETE_PROCESSING action 1`] = `
Object {
  "fetching": false,
  "processing": false,
}
`;

exports[`Job Reducer should handle ON_CREATE_JOB action 1`] = `
Object {
  "fetching": false,
  "processing": true,
}
`;

exports[`Job Reducer should handle ON_ERROR action 1`] = `
Object {
  "fetching": false,
  "job": Object {
    "status": "Failed",
  },
  "processing": false,
}
`;

exports[`Job Reducer should handle ON_GET_STATUS action 1`] = `
Object {
  "fetching": true,
  "processing": false,
}
`;

exports[`Job Reducer should handle ON_SUCCESS action 1`] = `
Object {
  "fetching": false,
  "job": Object {
    "id": "123",
    "status": "Created",
  },
  "processing": false,
}
`;

exports[`Job Reducer should handle ON_SUCCESS_GET_STATUS action 1`] = `
Object {
  "fetching": false,
  "job": Object {
    "status": Object {
      "status": "Processing",
    },
  },
  "processing": false,
}
`;
